arch/xtensa/Kconfig

```diff
@@ -93,6 +93,24 @@ config XTENSA_NUM_SPIN_RELAX_NOPS
 	  Specify the number of NOPs in Xtensa specific
 	  arch_spin_relax().
 
+config CPU_HAS_DSP
+	bool
+	help
+	  This option is enabled when the CPU has a hardware DSP unit such
+	  as a HiFi4.
+
+if CPU_HAS_DSP
+menu "DSP Options"
+
+config DSP_SHARING
```

Comment:
There is no specific HIFI DSP classification, @peter-mitsis , so there is no difference for HIFI3/4/5 for context save/restore?

Reply:
I wonder if CPU_HAS_XTENSA_HIFI would be a better kconfig name since this is about storing hifi registers?

Reply:
Your idea seems to align well with  #67513--happy to move in that direction. :)